tests/js/spec/components/activity/note/input.spec.jsx

```diff
@@ -104,7 +104,7 @@ describe('NoteInput', function() {
 
       expect(onUpdate).toHaveBeenCalledWith(
         {text: 'new item'},
-        {data: {text: 'an existing item'}}
+        {id: 'item-id', data: {text: 'an existing item'}}
```

Comment:
We need `id` now because we use it to determine if the note is an existing item (update vs create).

We keep `data` here because the Input component just passes back the `item` prop so that its parent component (in this case, GroupActivity) can decide how to use it.

Reply:
If you passed up `id` and `text` it seems like it could still all work and you'd be shipping smaller structures around.